src/main/flight/failsafe.c

```diff
@@ -382,8 +382,10 @@ void failsafeUpdateState(void)
                     uint8_t failsafe_procedure_to_use = failsafeConfig()->failsafe_procedure;
 
                     // Craft is closer than minimum failsafe procedure distance (if set to non-zero)
+                    // GPS must also be working, and home position set
                     if ((failsafeConfig()->failsafe_min_distance > 0) && 
-                        (GPS_distanceToHome < failsafeConfig()->failsafe_min_distance)) {
+                        (GPS_distanceToHome < failsafeConfig()->failsafe_min_distance) &&
+                        navigationPositionEstimateIsHealthy()) {
```

Comment:
This will work even when navigation system is not healty. 
Check for `sensors(SENSOR_GPS) && STATE(GPS_FIX) && STATE(GPS_FIX_HOME)` - this should be enough to make sure we have valid GPS fix and home position is recorded.

Reply:
Ah, ok. I've made these changes.